compiler/src/iree/compiler/Dialect/Flow/Transforms/FuseDequantizationMatmul.cpp

```diff
@@ -43,8 +43,16 @@ static LogicalResult fuseDequantAndMatmul(RewriterBase &rewriter,
     regionOp = maybeRegionOp.value();
   }
 
-  FailureOr<DispatchRegionOp> maybeFusedRegionOp =
-      movePrecedingOpsIntoDispatchRegion(rewriter, dequant, regionOp);
+  FailureOr<DispatchRegionOp> maybeFusedRegionOp;
+  if (dequant->hasOneUse()) {
+    maybeFusedRegionOp =
+        movePrecedingOpsIntoDispatchRegion(rewriter, dequant, regionOp);
+  } else {
+    // Clone the dequant operation if there are multiple uses of dequant.
+    maybeFusedRegionOp =
+        clonePrecedingOpIntoDispatchRegion(rewriter, dequant, regionOp);
+  }
```

Comment:
I didn't think we'd need to differentiate the cases here. Maybe run CSE in the test? We'd want it to work like that anyway if a dequant has multiple (and only) matmul users.

Reply:
This is the question of whether should we leave DCE to other passes or should we do it ourselves. I think I prefer to do simple cleanups if they don't take much effort. I don't see what we would gain by not differentiating the cases here, so I prefer the different cases.

I don't have strong feelings about this, and if you still feel we need to change it, I can do it.

Reply:
We could make the dispatch also return the dequant result.